### PR TITLE
[WIP] UserRefactor

### PR DESCRIPTION
**Pull request by JackyChiu**

### CODE REVIEW

**Comment by guillaumemaka** [comment]
Maybe we should move all the AddRoute declaration in main.go and pass the router to ListenAndServe ?

**Comment by JackyChiu** [comment]
> Maybe we should move all the AddRoute declaration in main.go and pass the router to ListenAndServe ?

@guillaumemaka I like the idea of that since our "router builder" methods aren't really doing any specific logic. I'll do it in another PR

Another thing I'd like is to do is abstract the router to its own package (do you think middleware should go along with it?)

**Comment by guillaumemaka** [comment]
make sense to abstract the router and for middlewares we need a way to pass the DB and JWT interface that's currently available through the Handler struct, or maybe we could move DB, JWT and Logger to the router struct or make them available globally (not sure it's a good approach)

File: models/user.go [hunk 1]
```diff
@@ -32,9 +32,19 @@ func EncryptPassword(password string) string {
...
+	if email == "" {
+		errs["email"] = []string{EMPTY_MSG}
+	}
+	if username == "" {
+		errs["username"] = []string{EMPTY_MSG}
+	}
+	if password == "" {
+		errs["password"] = []string{EMPTY_MSG}
+	}
+	if errs != nil {
```
Comment (guillaumemaka) [inline models/user.go:46]: Will always evaluate to true try:

```go
if (len(errs["email"]) > 0) || (len(errs["username"]) > 0) || (len(errs["password"]) > 0) {
	return nil, errs
}
```

Comment (JackyChiu) [reply models/user.go:46]: Nice catch, this would be cleaner:
```go
if len(errs) > 0 {
  return nil, errs
}
```

**Comment by guillaumemaka** [comment]
👍

**Comment by JackyChiu** [comment]
@guillaumemaka this is still a "work in progress" btw

**Comment by guillaumemaka** [comment]
@JackyChiu I know. What's your current progress ? I'm currently working on the comments routes and the Gin implementation.

**Comment by JackyChiu** [comment]
I've been cleaning some stuff up, and will finish the users route sometime later this week or weekend

**Pull request closed without merging**